### PR TITLE
[DUOS-2471][risk=no] Consent: Implement POST `/api/dataset/index`

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2471

### Summary

Add admin-only `/api/dataset/index` and `/api/dataset/index/{datasetId}` POST endpoints which trigger the indexing of datasets in the system.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/assets/paths/datasetIndex.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+post:
+  summary: Index all datasets
+  description: Index all datasets in the system
+  tags:
+    - Dataset
```
Comment (rushtong) [inline src/main/resources/assets/paths/datasetIndex.yaml:6]: One request - can we add the `- Admin` tag here too so it shows up in the admin section? Same for index by id.

**Review by rushtong — APPROVED** [review]
One optional request - looks and works perfectly, thank you! 👍🏽

**Review by rushtong — APPROVED** [review]
👍🏽

**Review by kaitlynoflaherty — APPROVED** [review]
looks good

**Pull request merged**